R/dependencies.R

```diff
@@ -540,25 +540,35 @@ renv_dependencies_discover_description <- function(path,
       renv_restore_state(key = "root") %||%
       renv_project_resolve()
 
-    # get fields from settings
+    # if this appears to be the DESCRIPTION associated with the active project,
+    # and an explicit set of dependencies was provided in install, then use those
+    primary <- renv_path_same(file.path(project, "DESCRIPTION"), path)
+
+    # by default, respect fields defined in settings
     fields <- settings$package.dependency.fields(project = project)
 
-    # if this is the DESCRIPTION file for the active project, include
-    # the dependencies for the active profile (if any) and Suggested fields.
-    # collect profile-specific dependencies as well
-    if (renv_path_same(file.path(project, "DESCRIPTION"), path)) {
-      fmt <- "Config/renv/profiles/%s/dependencies"
-      profile <- renv_profile_get()
-      fields <- c(fields, "Suggests", sprintf(fmt, profile))
+    # check for overrides if this is the project's own DESCRIPTION file
+    if (primary) {
+      default <- the$install_dependency_fields %||% c(fields, "Suggests")
+      profile <- sprintf("Config/renv/profiles/%s/dependencies", renv_profile_get())
+      fields <- c(default, profile)
     }
 
+    # return computed fields
     fields
 
   }
 
   # make sure dependency fields are expanded
   fields <- renv_description_dependency_fields_expand(fields)
 
+  # propagate 'extra' dependency fields if requested in install
+  if (!is.null(the$install_dependency_fields)) {
```

Comment:
This only affects non-primary description fields, right?

Also the `dependencies` argument of `install` is documented to accept "strong" etc so I think you need another `renv_description_dependency_fields_expand(fields)` somewhere.

Reply:
Yeah, exactly. The idea being if you wanted to install all `Config/Needs/blah` packages, we'd want to install all discovered dependencies even from recursive dependencies.

Reply:
Hmm, possibly... part of the reason I had it outside that block was because of this:

https://github.com/rstudio/renv/blob/75c75dc51071a1a916ace5ec6f00ccfac6cd4b51/R/retrieve.R#L1033

In effect, I wanted to make sure that the 'extra' fields were passed along even if the caller supplied `fields = <...>` in some call to `renv_dependencies_discover_description()`.

Reply:
Also here: https://github.com/rstudio/renv/blob/75c75dc51071a1a916ace5ec6f00ccfac6cd4b51/R/package.R#L288-L292

Reply:
Overall, my feeling is that it's probably worth refactoring this function (and testing just the fields outputs), but I don't think that needs to happen in this PR.